apisix/plugins/body-transformer.lua

```diff
@@ -0,0 +1,198 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core              = require("apisix.core")
+local xml2lua           = require("xml2lua")
+local xmlhandler        = require("xmlhandler.tree")
+local template          = require("resty.template")
+local ngx               = ngx
+local decode_base64     = ngx.decode_base64
+local req_set_body_data = ngx.req.set_body_data
+local str_format        = string.format
+local type              = type
+local pcall             = pcall
+local pairs             = pairs
+
+
+local transform_schema = {
+    type = "object",
+    properties = {
+        input_format = { type = "string", enum = {"xml", "json"} },
+        template = { type = "string" },
+    },
+    required = {"template"},
+}
+
+local schema = {
+    type = "object",
+    properties = {
+        request = transform_schema,
+        response = transform_schema,
+    },
+}
+
+
+local _M = {
+    version  = 0.1,
+    priority = 1080,
+    name     = "body-transformer",
+    schema   = schema,
+}
+
+
+-- luacheck: ignore
+function string.escape_xml(s)
```

Comment:
luacheck has no inline option to depress "setting undefine field of global" warning. But depress the whole lua file is not proper. And `escape_*` has no conflict so far, so I think it's ok to use these names.

Reply:
The solution is not about making luacheck happy, but limiting the scope of the injection. If we don't deregister the methods, it will affect other modules including those not written by us.
Anyway, it is not a good idea to allow the monkey patch of the std lib outside our scope.

Reply:
Yes, I know. But setting methods on demand on the global object will make luacheck warns: `setting undefined field escape_xml of global string`, which could not be depressed via inline comment, so ignore the whole file in `.luacheckrc` instead?